homeassistant/components/dynalite/config_flow.py

```diff
@@ -31,28 +21,20 @@ def __init__(self):
     async def async_step_import(self, import_info):
         """Import a new bridge as a config entry."""
         LOGGER.debug("async_step_import - %s", import_info)
-        host = self.context[CONF_HOST] = import_info[CONF_HOST]
+        host = import_info[CONF_HOST]
+        await self.async_set_unique_id(host)
+        self._abort_if_unique_id_configured()
+
+        bridge = DynaliteBridge(self.hass, host)
+        if not await bridge.async_setup():
```

Comment:
This will never be true.

Reply:
fixed this. it now is using try_connection() to try to connect and fails if not successful.

i did leave an override with a config parameter ("nowait"). The reason is that I view Home Assistant as an always-on server. If it reboots for any reason, I want it to be back and fully operational. If there is an event where it reboots and when it comes up the Dynalite gateway is down, I would like it to keep trying. Is there a way to make a component fail to load but have Home Assistant retry after some time? Otherwise, if I wouldn't leave the override, it would never go back to being operational without a manual reboot.

Reply:
The import flow will not finish on restart if the config entry for this host has been created once. So the scenario you describe won't impact the config flow.

From inside `async_setup_entry` in the component module we can raise `ConfigEntryNotReady` to retry later if the gateway is down.

Reply:
The retry time is between 5 and 80 seconds:
https://github.com/home-assistant/home-assistant/blob/588f2cd920f955578f97edcab180e1f08d3a14c0/homeassistant/config_entries.py#L223-L241